packages/components/src/components/header/header.ts

```diff
@@ -0,0 +1,158 @@
+ *
+ * @status stable
+ * @since 1.0
+ *
+ * @slot - The header's default content.
+ *
+ * @csspart main - The container that wraps the header's content.
+ *
+ * @cssproperty --sd-header-inner-width - width of the header content
+ * @cssproperty --sd-header-inner-max-width - max-width of the header content
+ * @cssproperty --sd-header-padding-top - padding-top of the header content
+ * @cssproperty --sd-header-padding-bottom - padding-bottom of the header content
+ * @cssproperty --sd-header-padding-x - padding-left and padding-right of the header content
+ */
+@customElement('sd-header')
+export default class SdHeader extends SolidElement {
+  @property({ reflect: true, type: Boolean }) fixed = false;
+  @property({ attribute: 'auto-spacing', reflect: true, type: Boolean }) autoSpacing = true;
+
+  private refHeaderElement?: HTMLElement;
+  private resizeObserver?: ResizeObserver;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.setBodySpacing = this.setBodySpacing.bind(this);
+    this.addResizeObserver = this.addResizeObserver.bind(this);
+  }
+
+  firstUpdated(): void {
+    this.setBodySpacing();
+    this.addResizeObserver();
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+      this.resizeObserver = undefined;
+    }
+  }
+
+  updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+
+    if (changedProperties.has('autoSpacing')) {
+      this.setBodySpacing();
+      if (this.autoSpacing) {
+        this.addResizeObserver();
+      } else if (this.resizeObserver) {
+        this.resizeObserver.disconnect();
+        this.resizeObserver = undefined;
+      }
+    }
+  }
+
+  @debounce(100)
+  private onResize(): void {
+    if (this.autoSpacing) {
+      this.setBodySpacing();
+    }
+  }
+
+  private addResizeObserver(): void {
+    this.resizeObserver = new ResizeObserver(() => {
+      this.onResize();
+    });
+
+    if (this.refHeaderElement) {
+      this.resizeObserver.observe(this.refHeaderElement);
+    }
+  }
+
+  private setBodySpacing(): void {
+    if (this.autoSpacing && this.refHeaderElement) {
+      document.body.style.paddingTop = `${this.refHeaderElement.clientHeight}px`;
+    } else {
+      document.body.style.paddingTop = '';
+    }
+  }
+
+  render() {
+    return html`
+      <header
+        class=${cx('w-screen bg-white relative', this.fixed && 'fixed-shadow w-screen')}
+        role="banner"
+        @slotchange=${this.handleSlotChange}
+      >
+        <div part="main" class="relative">
+          <slot></slot>
+        </div>
+      </header>
+    `;
+  }
+
+  private handleSlotChange(event: Event): void {
+    const slot = event.target as HTMLSlotElement;
+    if (slot.assignedElements().length > 0) {
+      this.refHeaderElement = event.currentTarget as HTMLElement;
+      this.setBodySpacing();
+      this.addResizeObserver();
```

Comment:
I don't think that we need more then resizeObserver. Please use the same observer.